src/main/java/tc/oc/pgm/rotation/Rotation.java

```diff
@@ -0,0 +1,130 @@
+package tc.oc.pgm.rotation;
+
+import com.google.common.base.Optional;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.bukkit.configuration.ConfigurationSection;
+import tc.oc.pgm.api.PGM;
+import tc.oc.pgm.map.PGMMap;
+
+public class Rotation implements PGMMapOrderProvider {
+  private ConfigurationSection configurationSection;
+
+  private String name;
+  private boolean enabled;
+  private List<PGMMap> maps = new ArrayList<>();
+  private int players;
+  private int position;
+
+  Rotation(ConfigurationSection configurationSection, String name) {
+    this.configurationSection = configurationSection;
+    this.name = name;
+    this.enabled = configurationSection.getBoolean("enabled");
+    this.players = configurationSection.getInt("players");
+    this.position = configurationSection.getInt("position");
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  public boolean isEnabled() {
+    return enabled;
+  }
+
+  public void setEnabled(boolean enabled) {
+    this.enabled = enabled;
+  }
+
+  public List<PGMMap> getMaps() {
+    return Collections.unmodifiableList(maps);
+  }
+
+  public void setMaps(List<PGMMap> maps) {
+    this.maps = maps;
+  }
+
+  public int getPlayers() {
+    return players;
+  }
+
+  public void setPlayers(int players) {
+    this.players = players;
+  }
+
+  public void setPosition(int position) {
+    this.position = position;
+  }
+
+  public int getPosition() {
+    return position;
+  }
+
+  public void load() {
+    configurationSection
+        .getStringList("maps")
+        .forEach(
+            map -> {
+              final Optional<PGMMap> mapByNameOrId =
+                  PGM.GLOBAL.get().getMapLibrary().getMapByNameOrId(map);
+              if (mapByNameOrId.isPresent()) maps.add(mapByNameOrId.get());
+              else {
+                PGM.GLOBAL
+                    .get()
+                    .getLogger()
+                    .warning(
+                        "[Rotation] "
+                            + "["
+                            + name
+                            + "] "
+                            + map
+                            + " not found in map repo. Ignoring...");
+              }
+            });
+  }
+
+  private void rotate() {
+    if (position + 1 == maps.size() || position + 1 > maps.size()) {
+      position = 0;
+    } else {
+      movePosition(1);
+    }
+  }
+
+  public void movePosition(int positions) {
```

Comment:
Should probably be `private`

Reply:
The skip command uses it [here](https://github.com/BGMP/PGM/blob/c7dea5d5ace460df0e7362570ae493469ee28f13/src/main/java/tc/oc/pgm/commands/RotationCommands.java#L228). Should I make it so it does so somehow else?

Reply:
Oh, then never mind.

Reply:
I guess I'm marking this as resolved then qwq.